File: src/SOP8/Base/Account/Base.Account.IBLL/Request/RequestUserList.cs
using Response.Request;

namespace Base.Account.IBLL.Request
{
    public class RequestUserList : RequestSearchTextPage
    {
        private int? m_siteNo = null;

        private bool m_useTeamName = false;
        private bool m_useMemberName = false;
        private bool m_useNickName = false;
        private bool m_useJobLevel = false;
        private bool m_useJobPosition = false;
        private bool m_useGrade = false;
        private bool m_usePhoneNumber = false;
        private bool m_useEmail = false;
        private bool m_linkRegularMember = true;
        private int? m_sortType = null;
        private bool m_sortMethod = true;

        public int? SiteNo
        {
            get { return m_siteNo; }
            set { m_siteNo = value; }
        }

        public bool UseTeamName
        {
            get { return m_useTeamName; }
            set { m_useTeamName = value; }
        }

        public bool UseMemberName
        {
            get { return m_useMemberName; }
            set { m_useMemberName = value; }
        }

        public bool UseNickName
        {
            get { return m_useNickName; }
            set { m_useNickName = value; }
        }

        public bool UseJobLevel
        {
            get { return m_useJobLevel; }
            set { m_useJobLevel = value; }
        }

        public bool UseJobPosition
        {
            get { return m_useJobPosition; }
            set { m_useJobPosition = value; }
        }

        public bool UseGrade
        {
            get { return m_useGrade; }
            set { m_useGrade = value; }
        }

        public bool UsePhoneNumber
        {
            get { return m_usePhoneNumber; }
            set { m_usePhoneNumber = value; }
        }

        public bool UseEmail
        {
            get { return m_useEmail; }
            set { m_useEmail = value; }
        }

        // 연결된 정규조직원이 반드시 필요한가?
        public bool LinkRegularMember
        {
            get { return m_linkRegularMember; }
            set { m_linkRegularMember = value; }
        }

        // 정렬 대상 필드 구분값 (예: 0=멤버명, 1=사용자 ID, 2=등급).
        // null 이면 기존 기본 정렬(user_sn)을 사용한다.
        public int? SortType
        {
            get { return m_sortType; }
            set { m_sortType = value; }
        }

        // 정렬 방향 값.
        // true : 오름차순(ASC), false : 내림차순(DESC)
        public bool SortMethod
        {
            get { return m_sortMethod; }
            set { m_sortMethod = value; }
        }

        public RequestUserList()
        {
        }

        public RequestUserList(string strSearchText, int pageNo, int? pageRowCount, int? siteNo, bool useTeamName, bool useMemberName, bool useNickName, bool useJobLevel, bool useJobPosition, bool useGrade, bool usePhoneNumber, bool useEmail, int? sortType = null, bool sortMethod = true)
        {
            this.SearchText = strSearchText;
            this.PageNo = pageNo;
            this.PageRowCount = pageRowCount;
            this.SiteNo = siteNo;
            this.UseTeamName = useTeamName;
            this.UseMemberName = useMemberName;
            this.UseNickName = useNickName;
            this.UseJobLevel = useJobLevel;
            this.UseJobPosition = useJobPosition;
            this.UseGrade = useGrade;
            this.UsePhoneNumber = usePhoneNumber;
            this.UseEmail = useEmail;
            this.SortType = sortType;
            this.SortMethod = sortMethod;
        }
    }
}

File: src/SOP8/광양/EgovLogin/base-account-ibll/src/main/java/egovframework/base/account/ibll/request/RequestUserList.java
package egovframework.base.account.ibll.request;

import egovframework.base.response.request.RequestSearchTextPage;

public class RequestUserList extends RequestSearchTextPage {

    private Integer siteNo = null;
    private boolean useTeamName = false;
    private boolean useMemberName = false;
    private boolean useNickName = false;
    private boolean useJobLevel = false;
    private boolean useJobPosition = false;
    private boolean useGrade = false;
    private boolean usePhoneNumber = false;
    private boolean useEmail = false;
    private boolean linkRegularMember = true;

    public Integer getSiteNo() {
        return siteNo;
    }

    public void setSiteNo(Integer siteNo) {
        this.siteNo = siteNo;
    }

    public boolean isUseTeamName() {
        return useTeamName;
    }

    public void setUseTeamName(boolean useTeamName) {
        this.useTeamName = useTeamName;
    }

    public boolean isUseMemberName() {
        return useMemberName;
    }

    public void setUseMemberName(boolean useMemberName) {
        this.useMemberName = useMemberName;
    }

    public boolean isUseNickName() {
        return useNickName;
    }

    public void setUseNickName(boolean useNickName) {
        this.useNickName = useNickName;
    }

    public boolean isUseJobLevel() {
        return useJobLevel;
    }

    public void setUseJobLevel(boolean useJobLevel) {
        this.useJobLevel = useJobLevel;
    }

    public boolean isUseJobPosition() {
        return useJobPosition;
    }

    public void setUseJobPosition(boolean useJobPosition) {
        this.useJobPosition = useJobPosition;
    }

    public boolean isUseGrade() {
        return useGrade;
    }

    public void setUseGrade(boolean useGrade) {
        this.useGrade = useGrade;
    }

    public boolean isUsePhoneNumber() {
        return usePhoneNumber;
    }

    public void setUsePhoneNumber(boolean usePhoneNumber) {
        this.usePhoneNumber = usePhoneNumber;
    }

    public boolean isUseEmail() {
        return useEmail;
    }

    public void setUseEmail(boolean useEmail) {
        this.useEmail = useEmail;
    }

    public boolean isLinkRegularMember() {
        return linkRegularMember;
    }

    public void setLinkRegularMember(boolean linkRegularMember) {
        this.linkRegularMember = linkRegularMember;
    }

    public RequestUserList() {
    }

    public RequestUserList(String searchText, int pageNo, Integer pageRowCount, Integer siteNo,
                            boolean useTeamName, boolean useMemberName, boolean useNickName,
                            boolean useJobLevel, boolean useJobPosition, boolean useGrade,
                            boolean usePhoneNumber, boolean useEmail) {
        this.setSearchText(searchText);
        this.setPageNo(pageNo);
        this.setPageRowCount(pageRowCount);
        this.siteNo = siteNo;
        this.useTeamName = useTeamName;
        this.useMemberName = useMemberName;
        this.useNickName = useNickName;
        this.useJobLevel = useJobLevel;
        this.useJobPosition = useJobPosition;
        this.useGrade = useGrade;
        this.usePhoneNumber = usePhoneNumber;
        this.useEmail = useEmail;
    }
}